internal/controller/kyma/controller.go

```diff
@@ -209,6 +207,42 @@ func (r *Reconciler) reconcile(ctx context.Context, kyma *v1beta2.Kyma) (ctrl.Re
 	return res, err
 }
 
+func (r *Reconciler) replaceKymaFromRemote(ctx context.Context, kyma *v1beta2.Kyma) ReconciliationLoopResult {
```

Comment:
to be honest, I don't think encapsulating the following logic into `replaceKymaFromRemote` is a good idea, first, it violates single responsibility principle. `SyncRemoteCrd` is nothing to do with `replaceKymaFromRemote` secondly, you complicated the result to return. 

I think it's perfectly fine keep  `SyncRemoteCrds` and `replaceSpecFromRemote` separated as before, and you don't have to introduce such `ReconciliationLoopResult` struct only for data transfer purpose, besides, the usage is only scoped in this function, I don't see the value.

In general, I don't like the idea we have multiple layer to transfer the `ctrl.Result, error`, this better always keep in the reconcile logic so that clearly indicate us in which step we stop reconciliation with what behavior (requeue need? contains error? ...),

Reply:
Yes, I introduced this function only to get rid of linter errors that complexity of other functions increased too much.
I cannot move this code back - because of it.
I would have to introduce a bigger refactoring which I wanted to avoid in THIS PR.
I have created [an issue to improve this](https://github.com/kyma-project/lifecycle-manager/issues/1750):

Reply:
I'd rather suppress linter if currently can't manage to make the `reconcile` logic shorter, but don't want to introduce such complicated encapuslation only for linter. but as I described here, I don't think you need to change this existing logic https://github.com/kyma-project/lifecycle-manager/pull/1748#discussion_r1711371963

Reply:
I'll try to not Validate Kyma Spec as you described in your other comment. If not possible, I'll move that logic back into main reconciliation loop and supress the linter.

Reply:
done.